ios-base/Onboarding/ViewModels/SignUpViewModel.swift

```diff
@@ -9,16 +9,20 @@
 import Foundation
 import UIKit
 
-protocol SignUpViewModelDelegate: class {
+protocol SignUpViewModelDelegate: SignInStateDelegate {
   func formDidChange()
-  func didUpdateState()
+}
+
+enum SignInViewModelState {
```

Comment:
Given that you are reusing this in multiple view models I'd name it `AuthViewModelState` (and `AuthViewModelStateDelegate` too), and move it to a separate file.